pelops/datasets/veri.py

```diff
@@ -34,31 +35,65 @@ class VeriDataset(chip.ChipDataset):
         "image_query",
         "image_test",
         "image_train",
-        "list_color.txt", 
+        "list_color.txt",
         "list_type.txt",
         "gt_image.txt",
         "jk_image.txt",
-        "train_label.txt"
+        "train_label.xml"
     )
 
     def __init__(self, dataset_path, set_type=None):
         super().__init__(dataset_path, set_type)
         self.__set_filepaths()  # set self.__filepaths
+        self.__color_type = {}
+        if self.set_type is utils.SetType.ALL or self.set_type is utils.SetType.TRAIN:
+            self.__build_metadata_dict()
         self.__set_chips()
 
+    def __build_metadata_dict(self):
+        """Extract car type and color from the label file."""
+        root = xml.etree.ElementTree.parse(self.__filepaths.label_train).getroot()
+
+        colors = {
+            1: "yellow", 2: "orange", 3: "green", 4: "gray", 5: "red",
+            6: "blue", 7: "white", 8: "golden", 9: "brown", 10: "black",
+        }
+        types = {
+            1: "sedan", 2: "suv", 3: "van", 4: "hatchback", 5: "mpv",
+            6: "pickup", 7: "bus", 8: "truck", 9: "estate",
+        }
+
+        self.__color_type = {}
+        for child in root.iter("Item"):
+            # Get the IDs from the XML node
+            vehicle_id = child.attrib.get("vehicleID", None)
+
+            if vehicle_id is not None:
+                color = child.attrib.get("colorID", None)
+                vehicle = child.attrib.get("typeID", None)
+
+                try:
+                    color_id = int(color)
+                    vehicle_id = int(vehicle)
+                except ValueError:
+                    continue
```

Comment:
should let ppl know that there was an exception..
except ValueError as ex:
...         print('color:{0} or vehicle:{1} could not be converted to int'.format(color,vehicle))

Is there a logger for the project?